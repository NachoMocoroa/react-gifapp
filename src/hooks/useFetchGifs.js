import { useState, useEffect } from 'react';
import { getAPIGifs } from '../helpers/getAPIGifs';

export const useFetchGifs = (category) => {

    const [ state, setState ] = useState({
        data: [],
        loading: true
    });
    
    useEffect(() => {
        getAPIGifs(category)
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                });
            })
    }, [category]);

    return state;
};
