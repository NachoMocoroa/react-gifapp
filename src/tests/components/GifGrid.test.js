import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test GifGrid', () => {
    
    const category = 'gif';

    test('should render GifGrid component', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render items on useFetchGifs load', () => {
        const gifs = [
            {
                id: '123',
                title: 'Title',
                url: 'https://localhost/image.gif',
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

});
