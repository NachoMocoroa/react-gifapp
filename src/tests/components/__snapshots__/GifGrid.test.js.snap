// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid should render GifGrid component 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    gif
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
